etna/transforms/timestamp/holiday.py

```diff
@@ -1,20 +1,51 @@
-import datetime
 from enum import Enum
 from typing import List
 from typing import Optional
 
 import holidays
 import numpy as np
 import pandas as pd
-
+from pandas.tseries.offsets import MonthBegin
+from pandas.tseries.offsets import MonthEnd
+from pandas.tseries.offsets import QuarterBegin
+from pandas.tseries.offsets import QuarterEnd
+from pandas.tseries.offsets import Week
+from pandas.tseries.offsets import YearBegin
+from pandas.tseries.offsets import YearEnd
+
+from etna.datasets import TSDataset
 from etna.transforms.base import IrreversibleTransform
 
 
+def bigger_than_day(freq: str):
+    """Compare frequency with day."""
+    dt = "2000-01-01"
+    dates_day = pd.date_range(start=dt, periods=2, freq="D")
+    dates_freq = pd.date_range(start=dt, periods=2, freq=freq)
+    return dates_freq[-1] > dates_day[-1]
+
+
+def define_period(offset: pd.tseries.offsets.BaseOffset, dt: pd.Timestamp, freq: str):
+    """Define start_date and end_date of period using dataset frequency."""
+    if isinstance(offset, (Week, MonthEnd, QuarterEnd, YearEnd)):
```

Comment:
I think it works incorrectly with `W-MON` and other parameteric offsets, e.g. YearEnd. Look at [documentation](https://pandas.pydata.org/docs/user_guide/timeseries.html). I think we should fully support at least offsets that you set in this function.

Reply:
After this we should probably clarify which offsets do we support and tell people how to determine which offset do they use by their frequency (because there is an offset for each frequency).

Reply:
From my point of view this works correctly, but maybe i don’t fully understand the logic of calculating days off (what specific period we take). Can you give a test where my code gives incorrect results and show an example of correct behavior?